### PR TITLE
Add some additional exception handling to the scheduler

### LINKED ISSUE #2768: 500 error: Scheduler exception with Octopoes
image

### LINKED ISSUE #2769: Scheduler handle external services responses
See https://github.com/minvws/nl-kat-coordination/issues/2768

### PR DESCRIPTION
**Pull request by jpbruinsslot**
Changes:
* Added pydantic response model to octopoes. This will parse the paginated response from octopoes which will throw the correct `ValidationError` exception.
* Added `ExternalServicesException` that can be catched by the rest of the application and handle it correctly.

QA:

- Start kat
- Kill octopoes (docker-compose kill octopoes)
- Check scheduler logs
- Scheduler should not be terminated because of a faulty response from octopoes

Closes #2768
Closes #2769

### CODE REVIEW

File: mula/scheduler/connectors/errors.py [hunk 1]
```diff
@@ -12,7 +13,11 @@ def exception_handler(func):
     def inner_function(*args, **kwargs):
         try:
             return func(*args, **kwargs)
-        except pydantic.ValidationError as exc:
-            raise ValidationError("Not able to parse response from external service.") from exc
+        except httpx.ConnectError as exc:
+            raise ExternalServiceError("External service is not available.") from exc
+        except pydantic.ValidationError:
+            ExternalServiceError("Validation error occurred.")
```
Comment (underdarknl) [inline mula/scheduler/connectors/errors.py:19]: No raise here?

File: mula/scheduler/connectors/errors.py [hunk 2]
```diff
@@ -12,7 +13,11 @@ def exception_handler(func):
     def inner_function(*args, **kwargs):
         try:
             return func(*args, **kwargs)
-        except pydantic.ValidationError as exc:
-            raise ValidationError("Not able to parse response from external service.") from exc
+        except httpx.ConnectError as exc:
+            raise ExternalServiceError("External service is not available.") from exc
```
Comment (underdarknl) [inline mula/scheduler/connectors/errors.py:17]: I'd Subclass a ExternalConnectionError from the ExternalServiceError and raise raise it here. You can still catch the more generic ExternalServiceError if you dont hava a way to handle the more specific ExternalConnectionError.

File: mula/scheduler/connectors/errors.py (hunk 1 above)
Comment (jpbruinsslot) [reply mula/scheduler/connectors/errors.py:19]: good catch, thanks!

File: mula/scheduler/connectors/errors.py (hunk 2 above)
Comment (jpbruinsslot) [reply mula/scheduler/connectors/errors.py:17]: Yeah, that's helpful! Will also do that for the validation error.

**Review by ammar92 — APPROVED** [review]

**Review by ammar92 — APPROVED** [review]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [ ] I have managed to pass the onboarding flow
    - [ ] Objects and Findings are created properly
    - [ ] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Seems to work. With and without Octopoes service running, I can see `scheduler.connectors.errors.ExternalServiceError` being caught without the Scheduler crashing.

### Bug or feature?:
Not a bug, but we should add some messaging to the `scheduler.connectors.errors.ExternalServiceError`, as it's unclear what really has happened, and inner exceptions are usually clear (e.g. a not found error on the external service or connection error are typically clear and useful for the end user to see)

**Review by underdarknl — APPROVED** [review]

**Pull request merged**